modules/appeals_api/app/services/appeals_api/uploads/document_validation.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require 'pdf_info'
+
+module AppealsApi
+  module Uploads
+    class DocumentValidation
+      def initialize(upload)
+        @document = upload[:document]
+      end
+
+      def validate
+        # TODO: log anything here if errors raised or log in the controller if fails validation?
+        raise UploadValidationError, non_pdf_error unless document_is_pdf?
+        raise UploadValidationError, max_dimension_error unless valid_page_dimensions?
+        raise UploadValidationError, file_size_error unless valid_file_size?
+      end
+
+      def document_is_pdf?
```

Comment:
This was interesting. The solution I finally came up with seems to account for both non-pdf files with non-pdf extensions, _and_ non-pdf files with false .pdf extensions.